File: app/src/main/java/com/spidchenko/week2task/repositories/SharedPrefRepository.java
package com.spidchenko.week2task.repositories;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;
import android.util.Log;

public class SharedPrefRepository {
    private static final String PREF_FILE_KEY = "com.spidchenko.week2task.PREF_FILE_KEY";
    private static final String PREF_USER_ID_KEY = "com.spidchenko.week2task.PREF_LOGIN_KEY";
    private static final String PREF_LAST_SEARCH_KEY = "com.spidchenko.week2task.PREF_LAST_SEARCH_KEY";
    private static final String TAG = "SharedPrefRepo.LOG_TAG";

    private static volatile SharedPrefRepository sInstance;
    private static SharedPreferences sSharedPreferences;


    private SharedPrefRepository(final Application application) {
        sSharedPreferences = application.getSharedPreferences(PREF_FILE_KEY, Context.MODE_PRIVATE);
    }

    public static SharedPrefRepository getInstance(final Application application) {
        if (sInstance == null) {
            synchronized (SharedPrefRepository.class) {
                if (sInstance == null) {
                    sInstance = new SharedPrefRepository(application);
                }
            }
        }
        return sInstance;
    }

    public int getUserId() {
        Log.d(TAG, "getLogin: " + sSharedPreferences.getInt(PREF_USER_ID_KEY, -1));
        return sSharedPreferences.getInt(PREF_USER_ID_KEY, -1);
    }

    public String getLastSearch() {
        Log.d(TAG, "getLastSearch: " + sSharedPreferences.getString(PREF_LAST_SEARCH_KEY, ""));
        return sSharedPreferences.getString(PREF_LAST_SEARCH_KEY, "");
    }

    public void saveUserId(int userId) {
        Log.d(TAG, "saveUserId: " + userId);
        SharedPreferences.Editor editor = sSharedPreferences.edit();
        editor.putInt(PREF_USER_ID_KEY, userId);
        editor.apply();
    }

    public void saveLastSearch(String lastSearch) {
        Log.d(TAG, "saveLastSearch: " + lastSearch);
        SharedPreferences.Editor editor = sSharedPreferences.edit();
        editor.putString(PREF_LAST_SEARCH_KEY, lastSearch);
        editor.apply();
    }
}

File: app/src/main/java/com/spidchenko/week2task/repositories/SharedPrefRepository.kt
package com.spidchenko.week2task.repositories

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import android.util.Log
import kotlin.jvm.Volatile

class SharedPrefRepository private constructor(application: Application) {
    val userId: Int
        get() {
            Log.d(TAG, "getLogin: " + sSharedPreferences.getInt(PREF_USER_ID_KEY, -1))
            return sSharedPreferences.getInt(PREF_USER_ID_KEY, -1)
        }
    val lastSearch: String?
        get() {
            Log.d(TAG, "getLastSearch: " + sSharedPreferences.getString(PREF_LAST_SEARCH_KEY, ""))
            return sSharedPreferences.getString(PREF_LAST_SEARCH_KEY, "")
        }

    fun saveUserId(userId: Int) {
        Log.d(TAG, "saveUserId: $userId")
        val editor = sSharedPreferences.edit()
        editor.putInt(PREF_USER_ID_KEY, userId)
        editor.apply()
    }

    fun saveLastSearch(lastSearch: String) {
        Log.d(TAG, "saveLastSearch: $lastSearch")
        val editor = sSharedPreferences.edit()
        editor.putString(PREF_LAST_SEARCH_KEY, lastSearch)
        editor.apply()
    }

    companion object {
        private const val PREF_FILE_KEY = "com.spidchenko.week2task.PREF_FILE_KEY"
        private const val PREF_USER_ID_KEY = "com.spidchenko.week2task.PREF_LOGIN_KEY"
        private const val PREF_LAST_SEARCH_KEY = "com.spidchenko.week2task.PREF_LAST_SEARCH_KEY"
        private const val TAG = "SharedPrefRepo.LOG_TAG"

        @Volatile
        private var sInstance: SharedPrefRepository? = null
        private lateinit var sSharedPreferences: SharedPreferences

        @JvmStatic
        fun getInstance(application: Application): SharedPrefRepository? {
            if (sInstance == null) {
                synchronized(SharedPrefRepository::class.java) {
                    if (sInstance == null) {
                        sInstance = SharedPrefRepository(application)
                    }
                }
            }
            return sInstance
        }
    }

    init {
        sSharedPreferences = application.getSharedPreferences(PREF_FILE_KEY, Context.MODE_PRIVATE)
    }
}